packages/hnt-to-rent-service/src/jupiter.ts

```diff
@@ -0,0 +1,184 @@
+        key: new PublicKey(addressLookupTableAddress),
+        state: AddressLookupTableAccount.deserialize(accountInfo.data),
+      });
+      acc.push(addressLookupTableAccount);
+    }
+
+    return acc;
+  }, new Array<AddressLookupTableAccount>());
+};
+
+const buildJupiterApi = (() => {
+  let api: DefaultApi | undefined;
+
+  return () => {
+    if (api) return api;
+    const config = new Configuration({ basePath: JUPITER_URL });
+    api = new DefaultApi(config);
+    return api;
+  };
+})();
+
+export const estimate = async ({
+  mint,
+}: {
+  mint: PublicKey;
+}): Promise<string> => {
+  if (
+    ![
+      HNT_MINT.toBase58(),
+      IOT_MINT.toBase58(),
+      MOBILE_MINT.toBase58(),
+    ].includes(mint.toBase58())
+  ) {
+    throw new Error(
+      `Provided mint not supported by service: ${mint.toBase58()}`
+    );
+  }
+
+  const connection = provider.connection;
+  const decimals = (await getMint(connection, NATIVE_MINT)).decimals;
+  const client = buildJupiterApi();
+  const quote = await client.quoteGet({
+    amount: toBN(0.020015, decimals).toNumber(),
+    inputMint: NATIVE_MINT.toBase58(),
+    outputMint: mint.toBase58(),
+    slippageBps: 100, // 1%
+    platformFeeBps: Number(JUPITER_FEE_BPS) || 0,
+  });
+
+  if (!quote) throw new Error('Unable to quote');
+  return quote.outAmount;
+};
+
+export const fundFees = async ({
+  userWallet,
+  mint,
+}: {
+  userWallet: PublicKey;
+  mint: PublicKey;
+}): Promise<VersionedTransaction> => {
+  if (
+    ![
+      HNT_MINT.toBase58(),
+      IOT_MINT.toBase58(),
+      MOBILE_MINT.toBase58(),
+    ].includes(mint.toBase58())
+  ) {
+    throw new Error(
+      `Provided mint not supported by service: ${mint.toBase58()}`
+    );
+  }
+
+  const connection = provider.connection;
+  const platformWallet = provider.wallet;
+  const client = buildJupiterApi();
+  const inputAmount = await estimate({ mint });
+  const quote = await client.quoteGet({
+    amount: Number(inputAmount),
+    inputMint: mint.toBase58(),
+    outputMint: NATIVE_MINT.toBase58(),
+    slippageBps: 100, // 1%
+    platformFeeBps: Number(JUPITER_FEE_BPS) || 0,
+  });
+
+  // Tx contains instructions to create/close WSOL token account
+  const {
+    computeBudgetInstructions,
+    setupInstructions,
+    swapInstruction,
+    cleanupInstruction,
+    addressLookupTableAddresses,
+  } = await client.swapInstructionsPost({
+    swapRequest: {
+      quoteResponse: quote,
+      userPublicKey: userWallet.toBase58(),
+      feeAccount: JUPITER_FEE_ACCOUNT || undefined,
+    },
+  });
+
+  // create rent flash loan instructions
+  const ataRent = await connection.getMinimumBalanceForRentExemption(
+    ACCOUNT_SIZE
+  );
+
+  const borrowIx = SystemProgram.transfer({
+    fromPubkey: platformWallet.publicKey,
+    toPubkey: userWallet,
+    lamports: ataRent,
+  });
+
+  const repayIx = SystemProgram.transfer({
+    fromPubkey: userWallet,
+    toPubkey: platformWallet.publicKey,
+    lamports: ataRent,
```

Comment:
```suggestion
    lamports: ataRent + (10000),
```

^Force the user to also pay the transaction fee back into this wallet.

Reply:
ahh, good call